metrics/prometheus/prometheus_test.go

```diff
@@ -0,0 +1,344 @@
+		assertCounterVecValue(t, test.description, m.PutsBackend.PutBackendRequests, test.expInvalidCount, prometheus.Labels{FormatKey: InvFormatVal})
+		assertCounterVecValue(t, test.description, m.PutsBackend.PutBackendRequests, test.expDefTTLCount, prometheus.Labels{FormatKey: DefinesTTLVal})
+		assertCounterVecValue(t, test.description, m.PutsBackend.PutBackendRequests, test.expErrorCount, prometheus.Labels{FormatKey: ErrorVal})
+		assertHistogram(t, test.description, m.PutsBackend.RequestLength, test.expSizeHistCount, test.expSizeHistSum)
+	}
+}
+
+func TestConnectionMetrics(t *testing.T) {
+	testCases := []struct {
+		description                    string
+		testCase                       func(pm *PrometheusMetrics)
+		expectedOpenConnectionCount    float64
+		expectedAcceptConnectionErrors float64
+		expectedCloseConnectionErrors  float64
+	}{
+		{
+			description: "Add a connection to the open connection count",
+			testCase: func(pm *PrometheusMetrics) {
+				pm.IncreaseOpenConnections()
+			},
+			expectedOpenConnectionCount:    1,
+			expectedAcceptConnectionErrors: 0,
+			expectedCloseConnectionErrors:  0,
+		},
+		{
+			description: "Remove a connection from the open connection count",
+			testCase: func(pm *PrometheusMetrics) {
+				pm.DecreaseOpenConnections()
+			},
+			expectedOpenConnectionCount:    0,
+			expectedAcceptConnectionErrors: 0,
+			expectedCloseConnectionErrors:  0,
+		},
+		{
+			description: "Count a close connection error",
+			testCase: func(pm *PrometheusMetrics) {
+				pm.RecordCloseConnectionErrors()
+			},
+			expectedOpenConnectionCount:    0,
+			expectedAcceptConnectionErrors: 0,
+			expectedCloseConnectionErrors:  1,
+		},
+		{
+			description: "Count an accept connection error",
+			testCase: func(pm *PrometheusMetrics) {
+				pm.RecordCloseConnectionErrors()
+				pm.RecordAcceptConnectionErrors()
+			},
+			expectedOpenConnectionCount:    0,
+			expectedAcceptConnectionErrors: 1,
+			expectedCloseConnectionErrors:  2,
+		},
+	}
+
+	m := createPrometheusMetricsForTesting()
+
+	for _, test := range testCases {
+		test.testCase(m)
+
+		assertGaugeValue(t, test.description, m.Connections.ConnectionsOpened, test.expectedOpenConnectionCount)
+		assertCounterVecValue(t, test.description, m.Connections.ConnectionsErrors, test.expectedAcceptConnectionErrors, prometheus.Labels{ConnErrorKey: AcceptVal})
+		assertCounterVecValue(t, test.description, m.Connections.ConnectionsErrors, test.expectedCloseConnectionErrors, prometheus.Labels{ConnErrorKey: CloseVal})
+	}
+}
+
+func TestExtraTTLMetrics(t *testing.T) {
+	m := createPrometheusMetricsForTesting()
+
+	assertHistogram(t, "Assert the extra time to live in seconds allocated when metrics were created", m.ExtraTTL.ExtraTTLSeconds, 1, 5000.00)
+
+	m.RecordExtraTTLSeconds(5)
+	assertHistogram(t, "Assert the extra time to live in seconds was logged", m.ExtraTTL.ExtraTTLSeconds, 2, 5005.00)
+}
```

Comment:
Please add a gatekeeping test akin to TestMetricCountGatekeeping to ensure accidental high cardinality metrics are not added in the future.

Reply:
Added. Please let me know if you are ok with this much simpler approach compared to the `prebid-server` version